src/store/indexeddb-remote-worker.js

```diff
@@ -0,0 +1,104 @@
+/*
+Copyright 2017 Vector Creations Ltd
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+import q from "q";
+import LocalIndexedDBStoreBackend from "./indexeddb-local-backend.js";
+
+/**
+ * This class lives in the webworker and drives a LocalIndexedDbStoreBackend
+ * controlled by messages from the main process.
+ */
+class IndexedDbStoreWorker {
+    constructor(postMessage) {
```

Comment:
I'm unhappy with the level of documentation on this class. We don't seem to use it at all in this PR, but we do in a riot-web PR here: https://github.com/vector-im/riot-web/pull/3578/files#diff-34344b4ed4dd336568358240bfdfb066R26

We fail to mention that:
 - `onMessage` should be hooked into `window.onmessage`.
 - `postMessage` in the constructor should be `window.postMessage`.
 - This class should be instantiated in the script which will be fed into the `IndexedDBStore`'s `workerScript`.

Please give an example of the script (basically the entirety of https://github.com/vector-im/riot-web/pull/3578/files#diff-34344b4ed4dd336568358240bfdfb066R26 probably with fully-qualified `window.` for clarity rather than assumed global variables) in the constructor for this class.